lxd/networks.go

```diff
@@ -241,20 +241,32 @@ func networksGet(d *Daemon, r *http.Request) response.Response {
 
 	recursion := util.IsRecursionRequest(r)
 
-	var networkNames []string
+	// networks holds the network names of the managed and unmanaged networks. They are in two different slices so that
+	// we can perform access control checks differently.
+	var networks [2][]string
 
 	err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
 		// Get list of managed networks (that may or may not have network interfaces on the host).
-		networkNames, err = tx.GetNetworks(ctx, effectiveProjectName)
+		networks[0], err = tx.GetNetworks(ctx, effectiveProjectName)
```

Comment:
please can we have local constants for the magic numbers 0 and 1 so we dont end up with hard to read logic like this:

```go
		if !shared.ValueInSlice(iface.Name, networks[0]) {
				networks[1] = append(networks[1], iface.Name)
			}
```

Reply:
Yep sure. I wasn't sure about this either and contemplated using a map but it seemed overkill for something so simple. I'll add some constants.